qiskit/circuit/library/arithmetic/inverse_chebyshev.py

```diff
@@ -0,0 +1,298 @@
+        """Set the error tolerance.
+
+        Note that this may change the underlying quantum register, if the number of state qubits
+        changes.
+
+        Args:
+            epsilon: The new error tolerance.
+        """
+        if self._epsilon is None or epsilon != self._epsilon:
+            self._invalidate()
+            self._epsilon = epsilon
+
+            self._reset_registers(self.num_state_qubits)
+
+    @property
+    def constant(self) -> float:
+        r"""The constant :math:`C` in :math:`arcsin(C/x)`.
+
+                Returns:
+                    The constant :math:`C` in :math:`arcsin(C/x)`.
+        """
+        return self._constant
+
+    @constant.setter
+    def constant(self, constant: Optional[float]) -> None:
+        r"""Set the constant :math:`C` in :math:`arcsin(C/x)`.
+
+        Note that this may change the underlying quantum register, if the number of state qubits
+        changes.
+
+        Args:
+            constant: The new constant :math:`C` in :math:`arcsin(C/x)`.
+        """
+        if self._constant is None or constant != self._constant:
+            self._invalidate()
+            self._constant = constant
+
+            self._reset_registers(self.num_state_qubits)
+
+    @property
+    def num_state_qubits(self) -> int:
+        r"""The number of state qubits representing the state :math:`|x\rangle`.
+
+        Returns:
+            The number of state qubits.
+        """
+        return self._num_state_qubits
+
+    @num_state_qubits.setter
+    def num_state_qubits(self, num_state_qubits: Optional[int]) -> None:
+        """Set the number of state qubits.
+
+        Note that this may change the underlying quantum register, if the number of state qubits
+        changes.
+
+        Args:
+            num_state_qubits: The new number of qubits.
+        """
+        if self._num_state_qubits is None or num_state_qubits != self._num_state_qubits:
+            self._invalidate()
+            self._num_state_qubits = num_state_qubits
+
+            self._reset_registers(num_state_qubits)
+
+    def _reset_registers(self, num_state_qubits: Optional[int]) -> None:
+        if num_state_qubits:
+            qr_state = QuantumRegister(num_state_qubits)
+            qr_target = QuantumRegister(1)
+            self.qregs = [qr_state, qr_target]
+
+            if self._poly_r is not None:
+                num_ancillas = self._poly_r.num_ancillas
+                if num_ancillas > 0:
+                    self._ancillas = []
+                    qr_ancilla = AncillaRegister(num_ancillas)
+                    self.add_register(qr_ancilla)
+        else:
+            self.qregs = []
+
+    def _build(self):
+        """Build the circuit. The operation is considered successful when q_objective is
+        :math:`|1>`."""
+        # We perform the identity operation on [1,a].
+        # int(round()) necessary to compensate for computer precision.
+        if self.num_state_qubits:
```

Comment:
Although in 99% of the cases this is the same, it's just a little bit safer to explicitly check for None
```suggestion
        if self.num_state_qubits is not None:
```